### PR TITLE
Add custom objects to command menu search + use ilike for notes search

### LINKED ISSUE #8522: Feature Request: Add Quick Search (Ctrl + K) for All Objects by Record Text
## Scope & Context

**Example:** 

We are working on enabling users to use quick search (Ctrl + K) for all objects in their workspace.  
The ticket is about allowing search functionality for custom objects based on their record text in addition to standard objects.

## Current behavior

The quick search (Ctrl + K) functionality currently supports only standard objects (e.g., predefined types such as people, notes, or companies). Custom objects are not included in the search results, even if the record text matches the query.  

## Expected behavior

The quick search (Ctrl + K) should support both standard and custom objects. The search results should include any custom objects where the record text matches the query. And on click on the search result, user is redirected to detail page of the object

### PR DESCRIPTION
**Pull request by ijreilly**
In this PR

- Re-introduce previously used search based on "ILIKE" queries for search on notes since the tsvector search with json text is not working correctly (@charlesBochet)
- Add search on custom objects in Command Menu bar (closes https://github.com/twentyhq/twenty/issues/8522)

https://github.com/user-attachments/assets/0cc064cf-889d-4f2c-8747-6d8670f35a39

### CODE REVIEW

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx [hunk 1]
```diff
@@ -170,36 +173,52 @@ export const CommandMenu = () => {
...
-      skip: !isCommandMenuOpened,
-      objectNameSingular: CoreObjectNameSingular.Person,
-      limit: 3,
-      searchInput: deferredCommandMenuSearch ?? undefined,
-    });
+  const {
+    matchesSearchFilterObjectRecordsQueryResult,
+    matchesSearchFilterObjectRecordsLoading: loading,
+  } = useMultiObjectSearch({
+    excludedObjects: [CoreObjectNameSingular.Task, CoreObjectNameSingular.Note],
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:180]: maybe we actually want to include tasks?

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx [hunk 2]
```diff
@@ -170,36 +173,52 @@ export const CommandMenu = () => {
...
+  const { objectRecordsMap: matchesSearchFilterObjectRecords } =
+    useMultiObjectSearchQueryResultFormattedAsObjectRecordsMap({
+      multiObjectRecordsQueryResult:
+        matchesSearchFilterObjectRecordsQueryResult,
     });
 
-  const { loading: isNotesLoading, records: notes } = useSearchRecords<Note>({
+  const { loading: isNotesLoading, records: notes } = useFindManyRecords<Note>({
     skip: !isCommandMenuOpened,
     objectNameSingular: CoreObjectNameSingular.Note,
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:193]: Missing Task?

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx (hunk 1 above)
Comment (Weiko) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:180]: task is also using RICH_TEXT

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx [hunk 3]
```diff
@@ -170,36 +173,52 @@ export const CommandMenu = () => {
...
+  );
+  const opportunities = matchesSearchFilterObjectRecords.opportunities?.map(
+    (opportunities) => opportunities.record,
+  );
+
+  const customObjectRecordsMap = useMemo(() => {
+    return Object.fromEntries(
+      Object.entries(matchesSearchFilterObjectRecords).filter(
+        ([namePlural, records]) =>
+          !['people', 'opportunities', 'companies'].includes(namePlural) &&
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:217]: nit: use CoreObjectNameSingular

Comment (Weiko) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:217]: My bad, it's plural!

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx (hunk 2 above)
Comment (ijreilly) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:193]: added it !

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx [hunk 4]
```diff
@@ -170,36 +175,67 @@ export const CommandMenu = () => {
...
   });
 
-  const { loading: isOpportunitiesLoading, records: opportunities } =
-    useSearchRecords<Opportunity>({
-      skip: !isCommandMenuOpened,
-      objectNameSingular: CoreObjectNameSingular.Opportunity,
-      limit: 3,
-      searchInput: deferredCommandMenuSearch ?? undefined,
-    });
+  const { loading: isTasksLoading, records: tasks } = useFindManyRecords<Note>({
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:205]:
```suggestion
  const { loading: isTasksLoading, records: tasks } = useFindManyRecords<Task>({
```

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx [hunk 5]
```diff
@@ -572,6 +653,43 @@ export const CommandMenu = () => {
...
+                          id={task.id}
+                          Icon={IconCheckbox}
+                          key={task.id}
+                          label={task.title ?? ''}
+                          onClick={() => openActivityRightDrawer(task.id)}
+                        />
+                      </SelectableItem>
+                    ))}
+                  </CommandGroup>
+                  {Object.entries(customObjectRecordsMap).map(
```
Comment (Weiko) [inline packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:669]: I feel like we could simplify and reuse this loop for standard objects as well? (avoiding hardcoded company/task/etc)

**Review by Weiko — COMMENTED** [review]
LGTM!

**Review by Weiko — APPROVED** [review]

**Comment by FelixMalfait** [comment]
Sorry for the late reply @ijreilly  - yes tasks too!

File: packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx (hunk 5 above)
Comment (ijreilly) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx:669]: Ok, it still took 100lines to declare all objects as they have different Icon and label logics, but at least we arent repeating ourselves now !

Comment (Weiko) [reply packages/twenty-front/src/modules/command-menu/components/CommandMenu.tsx]: Better imho, thank you @ijreilly

**Pull request merged**